src/object_cmd.cpp

```diff
@@ -346,21 +348,90 @@ CommandCost CmdBuildObject(DoCommandFlag flags, TileIndex tile, ObjectType type,
 			/* This may never be constructed using this method. */
 			return CMD_ERROR;
 
-		default: // i.e. NewGRF provided.
+		default: { // i.e. NewGRF provided.
+			build_object_size = GB(spec->size, 0, 4) * GB(spec->size, 4, 4);
```

Comment:
In line 222-223 the width and height of the object already determined, (but a little differently), so it could be:
```suggestion
			build_object_size = size_x * size_y;
```